Lib/copy.py

```diff
@@ -84,22 +84,19 @@ def copy(x):
         return _copy_immutable(x)
 
     copier = getattr(cls, "__copy__", None)
-    if copier:
+    if copier is not None:
         return copier(x)
 
     reductor = dispatch_table.get(cls)
-    if reductor:
+    if reductor is not None:
         rv = reductor(x)
     else:
         reductor = getattr(x, "__reduce_ex__", None)
```

Comment:
`__reduce_ex__` could be absent in old style classes in Python 2. New style classes inherit it from `object`. Therefore this can be just `x.__reduce_ex__`.

Reply:
Changing this line to `x.__reduce_ex__` makes `test_copy_cant` fail:

```
======================================================================
ERROR: test_copy_cant (test.test_copy.TestCopy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/berker/projects/cpython/master/Lib/test/test_copy.py", line 91, in test_copy_cant
    self.assertRaises(copy.Error, copy.copy, x)
  File "/home/berker/projects/cpython/master/Lib/unittest/case.py", line 743, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/home/berker/projects/cpython/master/Lib/unittest/case.py", line 178, in handle
    callable_obj(*args, **kwargs)
  File "/home/berker/projects/cpython/master/Lib/copy.py", line 91, in copy
    reductor = x.__reduce_ex__
  File "/home/berker/projects/cpython/master/Lib/test/test_copy.py", line 88, in __getattribute__
    raise AttributeError(name)
AttributeError: __reduce_ex__
```

The test was added since the initial check-in of the module, so perhaps we can this leave as-is and add a note describing the behavior?

Reply:
Lets defer this change to a separate issue.

It changes the behavior for the case when `__reduce__` is defined, but `__reduce_ex__` is set to None. If this change is intentional, it is worth to the behavior of pickle (both implementations) the same change. Currently the Python implementation will call `__reduce__` in this case, but the C implementation will raise a TypeError. If both `__reduce__` and `__reduce_ex__` are set to None, the current Python implementation will raise a PicklingError. It would be better if all implementation will have the same behavior in corner cases like this.